internal/security/secretstoreclient/vault.go

```diff
@@ -272,3 +161,49 @@ func (vc *vaultClient) commonRequest(token *string, method string, url string, j
 	req.Header.Set("Content-Type", JSONContentType)
 	return vc.client.Do(req)
 }
+
+// processResponse takes the output of the commonRequest method
+// and performs standard processing based the the following:
+// operationDescription contains a message describing the operation in logs
+// expectedStatusCode is the expected status code for success
+// responseObject is the address of a JSON struct, or nil
+// If the response is HTTP 200 status and responseObject is not nil,
+// this function will attempt to decode the object
+// delegate is a callback function called to handle the decoded response
+func (vc *vaultClient) processResponse(resp *http.Response, responseError error,
+	operationDescription string, expectedStatusCode int, responseObject interface{},
+	delegate func() error) (statusCode int, err error) {
```

Comment:
this is a bit weird that the `delegate` function doesn't take any arguments even though it's effectively acting like a callback. 

Since the `delegate` function is only ever used one place, could you just drop the `delegate` argument entirely and use the code there instead? AFAICT, the only advantage to using this delegate function is that you have error checking inside this function instead of outside it, but to me that is outweighed by the weirdness of having a callback function that takes no arguments.

Reply:
It removes duplication for not only the error checking, but the error logging, and also the log message that the request was performed successfully.  I could in fact pass `responseObject` into the callback, but since I don't know the type I'd have to pass it as interface{} and then write a type assertion in the callback to cast it back to the proper type. I in fact did this without the type assertion and was surprised the callback didn't fail because of the interface{} type, because I found that was using a closure, and the callback already had access to the parsed json body and it had the right type.

In most cases turns a simple API call into a 3-liner

```
	parameters := RevokeTokenAccessorRequest{Accessor: accessor}
	resp, err := vc.commonJSONRequest(&token, http.MethodPost, RevokeAccessorAPI, parameters)
	return vc.processResponse(resp, err, "revoke token accessor", http.StatusNoContent, nil, noop)
```

or a 6-liner if there is something in the JSONresponse we want to keep
```
func (vc *vaultClient) LookupSelf(token string, tokenMetadata *TokenMetadata) (statusCode int, err error) {
	var response TokenLookupResponse
	resp, err := vc.commonRequest(&token, http.MethodGet, LookupSelfAPI, nil)
	return vc.processResponse(resp, err, "lookup self token", http.StatusOK, &response, func() error {
		*tokenMetadata = response.Data
		return nil
	})
}
```

Reply:
Addressed in #1855